src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala

```diff
@@ -105,11 +107,33 @@ class ScalafixInterface private (
       try arg(acc)
       catch { case NonFatal(e) => throw new InvalidArgument(e.getMessage) }
     }
-    new ScalafixInterface(newScalafixArguments, this.args ++ args)
+    new ScalafixInterface(
+      newScalafixArguments,
+      postScalafixHooks,
+      this.args ++ args
+    )
   }
 
   def run(): Seq[ScalafixError] =
-    scalafixArguments.run().toSeq
+    try {
+      scalafixArguments.run().toSeq
+    } finally {
+      val mayRewrite = args.reverse
+        .collectFirst {
+          case Arg.ParsedArgs(args) => args
+        }
+        .forall { args =>
+          Seq("--check", "--test", "--auto-suppress-linter-errors")
+            .intersect(args)
+            .isEmpty
+        }
+      if (mayRewrite) for {
+        hook <- postScalafixHooks
+        files <- args.reverse.collectFirst {
+          case Arg.Paths(paths) => paths.map(_.toFile)
+        }
+      } yield Try(hook(files))
```

Comment:
after implementing this, I am wondering if it's really relevant to pass which files might have changed, as scalafmt has its own targets and its own cache.